index.d.ts

```diff
@@ -1,38 +1,38 @@
-import { Middleware, MiddlewareAPI } from 'redux';
+import { Middleware, MiddlewareAPI, Action } from 'redux';
 import { Observable, ObservableInput } from 'rxjs/Observable';
 import { Scheduler } from 'rxjs/Scheduler';
 import { Operator } from 'rxjs/Operator';
 
-export declare class ActionsObservable<T> extends Observable<T> {
+export declare class ActionsObservable<T extends Action> extends Observable<T> {
   /**
    * Just like RxJS itself, we can't actually make this method always type-safe
    * because we would need non-final position spread params e.g.
    *   `static of<T>(...items: T, scheduler?: Scheduler): ActionsObservable<T>`
    * which isn't possible in either JavaScript or TypeScript. So instead, we
    * provide safe typing for up to 6 items, following by a scheduler.
    */
-  static of<T>(item1: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, item4: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, item4: T, item5: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, item4: T, item5: T, item6: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(...array: Array<T | Scheduler>): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, item4: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, item4: T, item5: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, item4: T, item5: T, item6: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(...array: Array<T | Scheduler>): ActionsObservable<T>;
 
-  static from<T>(ish: ObservableInput<T>, scheduler?: Scheduler): ActionsObservable<T>;
-  static from<T, R>(ish: ArrayLike<T>, scheduler?: Scheduler): ActionsObservable<R>;
+  static from<T extends Action>(ish: ObservableInput<T>, scheduler?: Scheduler): ActionsObservable<T>;
+  static from<T extends Action, R extends Action>(ish: ArrayLike<T>, scheduler?: Scheduler): ActionsObservable<R>;
 
   constructor(input$: Observable<T>);
-  lift<R>(operator: Operator<T, R>): ActionsObservable<R>;
-  ofType(...key: string[]): ActionsObservable<T>;
-  ofType(...key: any[]): ActionsObservable<T>;
+  lift<R extends Action>(operator: Operator<T, R>): ActionsObservable<R>;
+  lift<R>(operator: Operator<T, R>): Observable<R>;
```

Comment:
Yeah, unfortunately we can't completely constrain the type parameter `R` of an inherited method. But by providing the 2 overloads

```ts
lift<R extends Action>(operator: Operator<T, R>): ActionsObservable<R>;
lift<R>(operator: Operator<T, R>): Observable<R>;
```

you get the next best thing: whenever you use an `Operator<T, R>` where `R extends Action`, it will assume the first signature, so the result type will be `ActionsObservable<R>`. 

In general, if you want to add methods to observables which _don't_ require the assumption that `T extends Action`, would it not work to augment the definition of `Observable` itself, similar to how RxJS itself adds its operators piecemeal:

```ts
declare module 'rxjs/Observable' {
    interface Observable<T> {
        myObservableOperator: Foo;
    }
}
```

Then `ActionsObservable` is free to define overrides that specifically apply when `T extends Action` constraint, which is what it should be all about, right?